tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingTask.scala

```diff
@@ -25,12 +38,36 @@ trait FhirMappingSourceContext extends Serializable {
 /**
  * Context/configuration for one of the source of the mapping that will read the source data from file system
  *
- * @param path        File path to the source file  e.g. patients.csv
+ * @param path        File path to the source file or folder, e.g., "patients.csv" or "patients".
  * @param fileFormat  Format of the file (csv | json | parquet) if it can not be inferred from path e.g. csv
  * @param options     Further options for the format (Spark Data source options for the format e.g. For csv -> https://spark.apache.org/docs/latest/sql-data-sources-csv.html#data-source-option)
  */
 case class FileSystemSource(path: String, fileFormat:Option[String] = None, options:Map[String, String] = Map.empty[String, String], override val preprocessSql: Option[String] = None) extends FhirMappingSourceContext {
-  def sourceType:String = fileFormat.getOrElse(path.split('.').last)
+  /**
+   * Determines the source type based on the file format and path.
+   *
+   * @return The determined source type.
+   * @throws MissingFileFormatException If the file format is missing or invalid.
+   */
+  def sourceType: String = {
+    // the determined format
+    var format = fileFormat
+    // split the path into segments based on the period ('.')
+    val pathSegments = path.split('.')
+
+    // if the file format is empty and there are exactly two segments in the path,
+    // set the file format based on the last segment of the path
+    if (format.isEmpty && pathSegments.length == 2) {
+      format = Some(pathSegments.last)
+    }
```

Comment:
I think when the file name is patients.2.csv, this breaks because of `pathSegments.length == 2` check. Maybe we should split if path has "." char then get the last value.

Reply:
Updated the if statement accordingly.